wasmtime/_value.py

```diff
@@ -25,6 +37,63 @@ def _unintern(val: int) -> typing.Any:
     return Val._id_to_extern.get(val)
 
 
+def get_valtype_attr(ty: ValType) -> str:
+    return val_id2attr[wasm_valtype_kind(ty._ptr)]  # type: ignore
+
+
+def val_getter(store_id: int, val_raw: wasmtime_val_raw_t, attr: str) -> typing.Union[int, float, "wasmtime.Func", typing.Any]:
+    val = getattr(val_raw, attr)
+
+    if attr == 'externref':
+        ptr = ctypes.POINTER(wasmtime_externref_t)
+        if not val:
+            return None
+        ffi = ptr.from_address(val)
+        if not ffi:
+            return None
+        extern_id = wasmtime_externref_data(ffi)
+        return _unintern(extern_id)
+    elif attr == 'funcref':
+        if val == 0:
+            return None
+        f = wasmtime_func_t()
+        f.store_id = store_id
+        f.index = val
+        return wasmtime.Func._from_raw(f)
+    return val
+
+
+def val_setter(store_id: int, dst: wasmtime_val_raw_t, attr: str, val: "IntoVal") -> None:
+    casted: typing.Union[Any, int, float, None, wasmtime.Func]
+    if attr == 'externref':
+        if isinstance(val, Val) and val._raw and val._raw.kind == WASMTIME_EXTERNREF.value:
+            casted = ctypes.addressof(val._raw.of.externref)
+        else:
+            ex = Val.externref(val)._raw
+            if ex:
+                casted = ctypes.addressof(ex.of.externref)
+            else:
+                casted = 0
+    elif attr == 'funcref':
+        if isinstance(val, Val) and val._raw and val._raw.kind == WASMTIME_FUNCREF.value:
+            casted = val._raw.of.funcref.index
```

Comment:
I don't think that this is quite right because it's not copying over the `store_id`. Same for below this is only copying in the `index` and not the `store_id`. Additionally the store needs to be checked for this branch.